src/Microsoft.SqlTools.ServiceLayer/Management/Common/DataContainer.cs

```diff
@@ -1187,35 +1188,41 @@ internal static System.Security.SecureString BuildSecureStringFromPassword(strin
         private static XmlDocument CreateDataContainerDocument(ConnectionInfo connInfo, bool databaseExists)
         {
             string xml = string.Empty;
+            var serverNameNode = new XElement("servername");
+            serverNameNode.Value = connInfo.ConnectionDetails.ServerName.ToUpper(CultureInfo.InvariantCulture);
 
             if (!databaseExists)
             {
                 xml =
                 string.Format(@"<?xml version=""1.0""?>
                 <formdescription><params>
-                <servername>{0}</servername>
-                <connectionmoniker>{0} (SQLServer, user = {1})</connectionmoniker>
+                {0}
+                <connectionmoniker>{1} (SQLServer, user = {2})</connectionmoniker>
                 <servertype>sql</servertype>
-                <urn>Server[@Name='{0}']</urn>
+                <urn>Server[@Name='{1}']</urn>
```

Comment:
Were you going to create an XElement for the urn & connectionmoniker nodes? I think that would be safer